azavea/pfb-network-connectivity
# Feature/awf/bike infrastructure tiles#504

## Overview

This implements the features requested in #504:
- [x] Creating new tilemaker job
- [x] Adding tilemaker style file
- [x] Adding layer to detail map layer picker

### Demo

Screenshot of Fort Collins CO, with a visible implementation of each of the requested infrastructure components:
![screen shot 2017-07-05 at 17 13 35](https://user-images.githubusercontent.com/1818302/27915178-7ce0f42a-6233-11e7-8ec8-09c47eb48689.png)

Screenshot of the tile layer after generation via tilemaker container on my development machine for the Germantown neighborhood:
![screen shot 2017-07-06 at 09 18 05](https://user-images.githubusercontent.com/1818302/27915177-7cdfd734-6233-11e7-84f4-543f141c825b.png)

### Notes

The `generate_tiles_from_shapefile` function in `run_tilemaker.sh` was extended to take an additional style parameter, allowing multiple differently styled tile layers to be generated from a single source shapefile.

The diff is a bit muddy, as I split the original Mapbox Studio "combined styles" style project into three separate studio projects, one for each tile layer style we generate. This has a few advantages:
- Avoids bugs, where styles defined in the bike_infrastructure style would overwrite existing styles from the ways style
- Simplifies the copy/paste, such that you copy all `<style>` tags from a given studio project xml file to the matching xml file in the `tilemaker/styles` directory. Before, you had to verify you were copying the correct set of styles from the combined source Studio project xml.

I also switched the Mapbox Studio project datasource to Fort Collins, CO, as that ways file has examples of all of the bike infrastructure components we want to be styling. The original AZ shapefile did not. 

It might help to view the diff one commit at a time to get a better sense of the progression of changes. The first three commits are the changes in the project code that implement the new feature, the remainder are changes to the working files for Mapbox Studio.

## Testing Instructions

- `docker-compose build tilemaker`
- Create and run a new local analysis job
- Create and run a new local tilemaker job
- Go to place detail view for the job you just ran
- Select "Bike Infrastructure" overlay in the map layer picker: The new overlay should appear, and the existing overlays should remain visually unchanged when selected.

Closes #504
